pgmpy/inference/ApproxInference.py

```diff
@@ -29,25 +29,38 @@ def __init__(self, model):
         self.model = model
 
     @staticmethod
-    def _get_factor_from_df(df, state_names):
+    def _get_factor_from_df(df, model_states):
```

Comment:
In my opinion, `state_names` is still a more suitable name here. I find `model_states` to be a bit confusing.